pkg/common/security/usergroup_test.go

```diff
@@ -93,64 +140,80 @@ func TestBrokenUserGroup(t *testing.T) {
 	if ug.Groups[0] != "100" {
 		t.Errorf("User 'testuser2' primary group resolved while it should not: %v", ug)
 	}
+	testCache.lock.Unlock()
 
 	ug, err = testCache.GetUserGroup("testuser3")
 	if err != nil {
 		t.Error("Lookup should not have failed: testuser3")
 	}
-	if len(testCache.ugs) != 2 {
-		t.Errorf("Cache not updated should have 2 entries %d", len(testCache.ugs))
-	}
-	// check returned info: 4 groups, primary group is duplicate
-	if len(ug.Groups) != 4 {
-		t.Errorf("User 'testuser3' not resolved correctly: duplicate primary group not filtered %v", ug)
-	}
+
+	assert.Equal(t, 2, testCache.getUGsize(), "Cache not updated should have 2 entries %d", len(testCache.ugs))
+	assert.Equal(t, 4, testCache.getUGGroupSize("testuser3"), "User 'testuser3' not resolved correctly: duplicate primary group not filtered %v", ug)
+
+	ug, err = testCache.GetUserGroup("unknown")
+	assert.ErrorContains(t, err, "lookup failed for user: unknown")
+
+	ug, err = testCache.GetUserGroup("testuser4")
+	assert.NilError(t, err)
+
+	ug, err = testCache.GetUserGroup("testuser5")
+	assert.ErrorContains(t, err, "lookup failed for user: testuser5")
+
+	ug, err = testCache.GetUserGroup("invalid-gid-user")
+	testCache.lock.Lock()
+	assert.ErrorContains(t, err, "lookup failed for user: invalid-gid-user")
+	exceptedGroup := []string{"1_001"}
+	assert.Assert(t, reflect.DeepEqual(ug.Groups, exceptedGroup), fmt.Errorf("group should be: %v, but got: %v", exceptedGroup, ug.Groups))
+	testCache.lock.Unlock()
 }
 
 func TestGetUserGroupFail(t *testing.T) {
 	testCache := GetUserGroupCache("test")
 	testCache.resetCache()
 	// test cache should be empty now
-	if len(testCache.ugs) != 0 {
-		t.Fatalf("Cache not empty: %v", testCache.ugs)
-	}
+	assert.Equal(t, 0, testCache.getUGsize(), "Cache is not empty: %v", testCache.getUGmap())
 
 	// resolve an empty user
 	ug, err := testCache.GetUserGroup("")
+	testCache.lock.Lock()
 	if err == nil {
 		t.Error("Lookup should have failed: empty user")
 	}
 	// ug is empty everything should be nil..
 	if ug.User != "" || len(ug.Groups) != 0 || ug.resolved != 0 || ug.failed {
 		t.Errorf("UserGroup is not empty: %v", ug)
 	}
+	testCache.lock.Unlock()
 
 	// resolve a non existing user
 	ug, err = testCache.GetUserGroup("unknown")
+	testCache.lock.Lock()
 	if err == nil {
 		t.Error("Lookup should have failed: unknown user")
 	}
 	// ug is partially filled and failed flag is set
 	if ug.User != "unknown" || len(ug.Groups) != 0 || !ug.failed {
 		t.Errorf("UserGroup is not empty: %v", ug)
 	}
+	testCache.lock.Unlock()
```

Comment:
This lock/unlock pair can be removed. `GetUserGroup()` returns a copy and internally uses a mutex to access the map.